docs/src/expressions.md

```diff
@@ -4,30 +4,202 @@ DocTestSetup = quote
 end
 ```
 
-Expressions
-===========
+# Expressions
 
-DRAFT: JuMP has multiple types of expressions: affine, quadratic, and nonlinear.
-Just talk about affine and quadratic here (see other section for nonlinear).
-Describe the basic data structures and ways to construct expressions
-(these are: constructors, operators, `add_to_expression!`, and macros).
+JuMP has three types of expressions: affine, quadratic, and nonlinear. These
+expressions can be inserted into constraints or into the objective. This is
+particularly useful if an expression is used in multiple places in the model.
 
+## Affine expressions
 
-Example code with doc tests:
+There are four ways of constructing an affine expression in JuMP: using the
+[`@expression`](@ref) macro, by operator overloading, using the `AffExpr`
+constructor, and via [`JuMP.add_to_expression!`](@ref).
 
+### Macros
+
+The recommended way to create an affine expression is via the
+[`@expression`](@ref) macro.
+
+```jldoctest affine_macro
+model = Model()
+@variable(model, x)
+@variable(model, y)
+ex = @expression(model, 2x + y - 1)
+
+# output
+
+2 x + y - 1
+```
+
+This expression can be used in the objective, or added to a constraint. For
+example:
+```jldoctest affine_macro
+@objective(model, Min, 2 * ex - 1)
+JuMP.objective_function(model)
+
+# output
+
+4 x + 2 y - 3
+```
+
+Just like variables and constraints, named expressions can also be created. For
+example
+```jldoctest
+model = Model()
+@variable(model, x[i = 1:3])
+@expression(model, expr[i=1:3], i * sum(x[j] for j in i:3))
+expr
+
+# output
+
+3-element Array{JuMP.GenericAffExpr{Float64,VariableRef},1}:
+ x[1] + x[2] + x[3]
+ 2 x[2] + 2 x[3]
+ 3 x[3]
+```
+
+### Operator overloading
+
+Expressions can also be created outside the macro. However, not that this is
+much slower that constructing an expression using the macro. This should only be
```

Comment:
It can be asymptotically slower if a `sum()` is present. Otherwise it's likely pretty similar.